---
title: "Convert tRNAscan-SE output to GRanges"
author: "Felix G.M. Ernst"
date: "`r Sys.Date()`"
package: tRNAscan2GRanges
abstract: >
  Example conversion of tRNAscan-SE output for sacCer3 to a GRanges object
output: 
  BiocStyle::html_document:
    toc: true
    toc_float: true
    df_print: paged
vignette: >
  %\VignetteIndexEntry{tRNAscan2GRanges}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: references.bib
---

```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown(css.files = c('custom.css'))
```

# Introduction

tRNAscan-SE [@Lowe.1997] allows for the prediction of tRNA genes in whole
genomes based on sequence and calculated structural features. Many tRNA
annotations in genomes, for example the reference genome sacCer3 of
Saccharomyces cerevisiae, link to information generated by tRNAscan-SE.
However not all available information from tRNAscan-SE end up in the genome
annotation. Amongst these are for the structural information, additional scores
and the information, whether the conserved CCA-end is encoded in the genomic
DNA. To work with this complete set of information, the tRNAscan-SE output can
be parsed into a more accessible format, such as GRanges object.

# Getting started

The default tRNAscan-SE output, wether from running tRNAscan-SE [@Lowe.1997]
locally or retrieving the output from gtRNADb [@Chan.2016], consist of formatted
text document containing individual text blocks per tRNA delimited by an empty
line.

```{r}
library(tRNAscan2GRanges)
sacCer3_file <- system.file("extdata", 
                           file = "sacCer3-tRNAs.ss.sort", 
                           package = "tRNAscan2GRanges")

# output for sacCer3
# Before
readLines(con = sacCer3_file, n = 7L)
```

# Converting to GRanges

To access the information in a BioC context the conversion to a GRanges object 
comes to mind. This task is performed by `tRNAscan2GRanges()`, which uses 
regular expressions to extract the information from the text blocks. 

```{r}
# output for sacCer3
# After
gr <- tRNAscan2GRanges::tRNAscan2GRanges(sacCer3_file)
head(gr, 2)
```

The result can be used directly in R or saved as gff3/fasta file for further
use, including processing the sequences for HTS read mapping or statistical
analysis on tRNA content of the analyzed genome.

```{r}
suppressMessages(library(rtracklayer, quietly = TRUE))
# Save tRNA sequences
Biostrings::writeXStringSet(gr$tRNA_seq, filepath = tempfile())
# convert DNAStringSet to character require for gff3 file saving
# to be GFF3 compliant use tRNAscan2GFF
gff <- tRNAscan2GRanges::tRNAscan2GFF(sacCer3_file)
rtracklayer::export.gff3(gff, con = tempfile())
```

# Visualization

```{r}
# tRNAscan-SE output for hg38
hg38_file <- system.file("extdata", 
                        file = "hg38-tRNAs.ss.sort", 
                        package = "tRNAscan2GRanges")
# tRNAscan-SE output for E. coli MG1655
eco_file <- system.file("extdata", 
                       file = "eschColi_K_12_MG1655-tRNAs.ss.sort", 
                       package = "tRNAscan2GRanges")
# import tRNAscan-SE files
gr_hg <- tRNAscan2GRanges::tRNAscan2GRanges(hg38_file)
gr_eco <- tRNAscan2GRanges::tRNAscan2GRanges(eco_file)

# get summary plots if ggplot2 is installed
grl <- GenomicRanges::GRangesList(Sce = gr, 
                                  Hsa = gr_hg, 
                                  Eco = gr_eco)
plots <- tRNAscan2GRanges::plottRNAscanSummary(grl)
```
```{r, fig.cap = "tRNA length."}
plots$length
```
```{r, fig.cap = "tRNAscan-SE scores."}
plots$tRNAscan_score
```
```{r, fig.cap = "tRNA GC content."}
plots$gc
```
```{r, fig.cap = "tRNAs with introns."}
plots$introns
```

# Session info

```{r}
sessionInfo()
```

# References