<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Importing a tRNAscan output file as a GRanges object — import.tRNAscanAsGRanges • tRNAscanImport</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Importing a tRNAscan output file as a GRanges object — import.tRNAscanAsGRanges" />
<meta property="og:description" content="The function import.tRNAscanAsGRanges will import a tRNAscan-SE output
file and return the information as a GRanges object. The reported 
intron sequences are spliced from the result by default, but can also 
returned as imported.
The function tRNAScan2GFF formats the output of 
import.tRNAscanAsGRanges to be GFF3 compliant.
tRNAscanID generates a unique tRNA ID, which is like the format used 
in the SGD annotation
t*AminoAcidSingleLetter*(*Anticodon*)*ChromosomeIdentifier**optionalNumberIfOnTheSameChromosome* 
Example: tP(UGG)L or tE(UUC)E1." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tRNAscanImport</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/tRNAscanImport.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/FelixErnst/tRNAscanImport/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Importing a tRNAscan output file as a GRanges object</h1>
    <small class="dont-index">Source: <a href='https://github.com/FelixErnst/tRNAscanImport/blob/master/R/tRNAscanImport-import.R'><code>R/tRNAscanImport-import.R</code></a></small>
    <div class="hidden name"><code>import.tRNAscanAsGRanges.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The function <code>import.tRNAscanAsGRanges</code> will import a tRNAscan-SE output
file and return the information as a GRanges object. The reported 
intron sequences are spliced from the result by default, but can also 
returned as imported.</p>
<p>The function <code>tRNAScan2GFF</code> formats the output of 
<code>import.tRNAscanAsGRanges</code> to be GFF3 compliant.</p>
<p><code>tRNAscanID</code> generates a unique tRNA ID, which is like the format used 
in the SGD annotation</p>
<p><code>t*AminoAcidSingleLetter*(*Anticodon*)*ChromosomeIdentifier**optionalNumberIfOnTheSameChromosome*</code></p> 
<p>Example: tP(UGG)L or tE(UUC)E1.</p>
    </div>

    <pre class="usage"><span class='fu'>import.tRNAscanAsGRanges</span>(<span class='kw'>input</span>, as.GFF3 = <span class='fl'>FALSE</span>, trim.intron = <span class='fl'>TRUE</span>)

<span class='fu'>tRNAscan2GFF</span>(<span class='kw'>input</span>)

<span class='fu'>tRNAscanID</span>(<span class='kw'>input</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>input</th>
      <td><ul>
<li><p><code>import.tRNAscanAsGRanges</code>: a tRNAscan-SE input file</p></li>
<li><p><code>tRNAscan2GFF</code>: a compatible <code>GRanges</code> object such as the 
output of <code>import.tRNAscanAsGRanges</code></p></li>
</ul></td>
    </tr>
    <tr>
      <th>as.GFF3</th>
      <td><p>optional logical for <code>import.tRNAscanAsGRanges</code>: returns 
a gff3 compatible GRanges object directly. (default: <code>as.GFF3 = FALSE</code>)</p></td>
    </tr>
    <tr>
      <th>trim.intron</th>
      <td><p>optional logical for <code>import.tRNAscanAsGRanges</code>: 
remove intron sequences. This changes the tRNA length reported. To retrieve
the original length fo the tRNA gene, use the <code>width()</code> function on the 
GRanges object. (default: <code>trim.intron = TRUE</code>)</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a GRanges object</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Chan, Patricia P., and Todd M. Lowe. 2016. “GtRNAdb 2.0: An Expanded Database
of Transfer Rna Genes Identified in Complete and Draft Genomes.” Nucleic
Acids Research 44 (D1): D184–9. doi:10.1093/nar/gkv1309.</p>
<p>Lowe, T. M., and S. R. Eddy. 1997. “TRNAscan-Se: A Program for Improved
Detection of Transfer Rna Genes in Genomic Sequence.” Nucleic Acids Research
25 (5): 955–64.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>gr</span> <span class='op'>&lt;-</span> <span class='fu'>import.tRNAscanAsGRanges</span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, 
                               file = <span class='st'>"yeast.tRNAscan"</span>, 
                               package = <span class='st'>"tRNAscanImport"</span>))
<span class='kw'>gff</span> <span class='op'>&lt;-</span> <span class='fu'>tRNAscan2GFF</span>(<span class='kw'>gr</span>)
<span class='fu'><a href='https://rdrr.io/r/base/identical.html'>identical</a></span>(<span class='kw'>gff</span>,<span class='fu'>import.tRNAscanAsGRanges</span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"extdata"</span>, 
                               file = <span class='st'>"yeast.tRNAscan"</span>, 
                               package = <span class='st'>"tRNAscanImport"</span>),
                               as.GFF3 = <span class='fl'>TRUE</span>))</div><div class='output co'>#&gt; [1] TRUE</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Felix G.M. Ernst.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


