<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Importing tRNAscan-SE output as `GRanges` • tRNAscanImport</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Importing tRNAscan-SE output as `GRanges`">
<meta property="og:description" content="tRNAscanImport">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tRNAscanImport</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/tRNAscanImport.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/FelixErnst/tRNAscanImport/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="tRNAscanImport_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Importing tRNAscan-SE output as <code>GRanges</code>
</h1>
                        <h4 class="author">Felix G.M. Ernst</h4>
            
            <h4 class="date">2020-07-18</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/FelixErnst/tRNAscanImport/blob/master/vignettes/tRNAscanImport.Rmd"><code>vignettes/tRNAscanImport.Rmd</code></a></small>
      <div class="hidden name"><code>tRNAscanImport.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Example of importing a tRNAscan-SE output for sacCer3 as a GRanges object</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>tRNAscan-SE <span class="citation">(Lowe and Eddy 1997)</span> can be used for prediction of tRNA genes in whole genomes based on sequence context and calculated structural features. Many tRNA annotations in genomes contain or are based on information generated by tRNAscan-SE, for example the current SGD reference genome sacCer3 for Saccharomyces cerevisiae. However, not all available information from tRNAscan-SE end up in the genome annotation. Among these are for example structural information, additional scores and the information, whether the conserved CCA-end is encoded in the genomic DNA. To work with this complete set of information, the tRNAscan-SE output can be parsed into a more accessible GRanges object using <code>tRNAscanImport</code>.</p>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<p>The default tRNAscan-SE output, either from running tRNAscan-SE <span class="citation">(Lowe and Eddy 1997)</span> locally or retrieving the output from the gtRNADb <span class="citation">(Chan and Lowe 2016)</span>, consist of a formatted text document containing individual text blocks per tRNA delimited by an empty line.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://github.com/FelixErnst/tRNAscanImport">tRNAscanImport</a></span>)
<span class="kw">yeast_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,
                          file = <span class="st">"yeast.tRNAscan"</span>,
                          package = <span class="st">"tRNAscanImport"</span>)

<span class="co"># output for sacCer3</span>
<span class="co"># Before</span>
<span class="fu"><a href="https://rdrr.io/r/base/readLines.html">readLines</a></span>(con = <span class="kw">yeast_file</span>, n = <span class="fl">7L</span>)
</pre></div>
<pre><code>## [1] "chrI.trna1 (139152-139254)\tLength: 103 bp"                                                                  
## [2] "Type: Pro\tAnticodon: TGG at 33-35 (139184-139186)\tScore: 62.1"                                             
## [3] "Possible intron: 37-67 (139188-139218)"                                                                      
## [4] "HMM Sc=37.90\tSec struct Sc=24.20"                                                                           
## [5] "         *    |    *    |    *    |    *    |    *    |    *    |    *    |    *    |    *    |    *    |  " 
## [6] "Seq: GGGCGTGTGGTCTAGTGGTATGATTCTCGCTTTGGGcgacttcctgattaaacaggaagacaaagcaTGCGAGAGGcCCTGGGTTCAATTCCCAGCTCGCCCC"
## [7] "Str: &gt;&gt;&gt;&gt;&gt;.&gt;..&gt;&gt;&gt;.........&lt;&lt;&lt;.&gt;&gt;&gt;&gt;&gt;......................................&lt;&lt;&lt;&lt;&lt;.....&gt;&gt;&gt;&gt;&gt;.......&lt;&lt;&lt;&lt;&lt;&lt;.&lt;&lt;&lt;&lt;&lt;."</code></pre>
</div>
<div id="importing-as-granges" class="section level1">
<h1 class="hasAnchor">
<a href="#importing-as-granges" class="anchor"></a>Importing as <code>GRanges</code>
</h1>
<p>To access the information in a BioC context the import as a GRanges object comes to mind. <code><a href="../reference/import.tRNAscanAsGRanges.html">import.tRNAscanAsGRanges()</a></code> performs this task by evaluating each text block using regular expressions.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="co"># output for sacCer3</span>
<span class="co"># After</span>
<span class="kw">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import.tRNAscanAsGRanges.html">import.tRNAscanAsGRanges</a></span>(<span class="kw">yeast_file</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">gr</span>, <span class="fl">2</span>)
</pre></div>
<pre><code>## GRanges object with 2 ranges and 18 metadata columns:
##       seqnames        ranges strand |        no tRNA_length   tRNA_type
##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;   &lt;integer&gt; &lt;character&gt;
##   [1]     chrI 139152-139254      + |         1          72         Pro
##   [2]     chrI 166267-166339      + |         2          73         Ala
##       tRNA_anticodon tRNA_anticodon.start tRNA_anticodon.end tRNAscan_score
##          &lt;character&gt;            &lt;integer&gt;          &lt;integer&gt;      &lt;numeric&gt;
##   [1]            TGG                   33                 35           62.1
##   [2]            TGC                   34                 36           76.0
##                      tRNA_seq                tRNA_str tRNA_CCA.end
##                &lt;DNAStringSet&gt;   &lt;DotBracketStringSet&gt;    &lt;logical&gt;
##   [1] GGGCGTGTGG...AGCTCGCCCC &lt;&lt;&lt;&lt;&lt;.&lt;..&lt;...&gt;&gt;&gt;.&gt;&gt;&gt;&gt;&gt;.        FALSE
##   [2] GGGCACATGG...GTTGCGTCCA &lt;&lt;&lt;&lt;.&lt;&lt;..&lt;...&gt;&gt;&gt;&gt;.&gt;&gt;&gt;&gt;.        FALSE
##       tRNAscan_potential.pseudogene tRNAscan_intron.start tRNAscan_intron.end
##                           &lt;logical&gt;             &lt;integer&gt;           &lt;integer&gt;
##   [1]                         FALSE                139188              139218
##   [2]                         FALSE                  &lt;NA&gt;                &lt;NA&gt;
##       tRNAscan_intron.locstart tRNAscan_intron.locend tRNAscan_hmm.score
##                      &lt;integer&gt;              &lt;integer&gt;          &lt;numeric&gt;
##   [1]                       37                     67               37.9
##   [2]                     &lt;NA&gt;                   &lt;NA&gt;               53.4
##       tRNAscan_sec.str.score tRNAscan_infernal
##                    &lt;numeric&gt;         &lt;numeric&gt;
##   [1]                   24.2                NA
##   [2]                   22.6                NA
##   -------
##   seqinfo: 17 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co"># Any GRanges passing this, can be used for subsequent function</span>
<span class="fu"><a href="../reference/istRNAscanGRanges.html">istRNAscanGRanges</a></span>(<span class="kw">gr</span>)
</pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>The result can be used directly in R or saved as gff3/fasta file for further use, including processing the sequences for HTS read mapping or statistical analysis on tRNA content of the analyzed genome.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">Biostrings</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">rtracklayer</span>)
<span class="co"># suppressMessages(library(rtracklayer, quietly = TRUE))</span>
<span class="co"># Save tRNA sequences</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">writeXStringSet</a></span>(<span class="kw">gr</span><span class="op">$</span><span class="kw">tRNA_seq</span>, filepath = <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>())
<span class="co"># to be GFF3 compliant use tRNAscan2GFF</span>
<span class="kw">gff</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import.tRNAscanAsGRanges.html">tRNAscan2GFF</a></span>(<span class="kw">gr</span>)
<span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/GFFFile-class.html">export.gff3</a></span>(<span class="kw">gff</span>, con = <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>())
</pre></div>
</div>
<div id="visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h1>
<p>The tRNAscan-SE information can be visualized using the <code>gettRNAFeaturePlots()</code> function of the <code>tRNA</code> package, returning a named list of ggplot2 plots, which can be plotted or further modified. Alternatively, <code>gettRNASummary()</code> returns the aggregated information for further use.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="co"># tRNAscan-SE output for hg38</span>
<span class="kw">human_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,
                          file = <span class="st">"human.tRNAscan"</span>,
                          package = <span class="st">"tRNAscanImport"</span>)
<span class="co"># tRNAscan-SE output for E. coli MG1655</span>
<span class="kw">eco_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,
                        file = <span class="st">"ecoli.tRNAscan"</span>,
                        package = <span class="st">"tRNAscanImport"</span>)
<span class="co"># import tRNAscan-SE files</span>
<span class="kw">gr_human</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import.tRNAscanAsGRanges.html">import.tRNAscanAsGRanges</a></span>(<span class="kw">human_file</span>)
<span class="kw">gr_eco</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import.tRNAscanAsGRanges.html">import.tRNAscanAsGRanges</a></span>(<span class="kw">eco_file</span>)

<span class="co"># get summary plots</span>
<span class="kw">grl</span> <span class="op">&lt;-</span> <span class="fu">GRangesList</span>(Sce = <span class="kw">gr</span>,
                   Hsa = <span class="kw">gr_human</span>,
                   Eco = <span class="kw">gr_eco</span>)
<span class="kw">plots</span> <span class="op">&lt;-</span> <span class="fu">gettRNAFeaturePlots</span>(<span class="kw">grl</span>)
</pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">length</span>
</pre></div>
<div class="figure">
<img src="tRNAscanImport_files/figure-html/plot1-1.png" alt="tRNA length." width="700"><p class="caption">
tRNA length.
</p>
</div>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">tRNAscan_score</span>
</pre></div>
<div class="figure">
<img src="tRNAscanImport_files/figure-html/plot2-1.png" alt="tRNAscan-SE scores." width="700"><p class="caption">
tRNAscan-SE scores.
</p>
</div>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">gc</span>
</pre></div>
<div class="figure">
<img src="tRNAscanImport_files/figure-html/plot3-1.png" alt="tRNA GC content." width="700"><p class="caption">
tRNA GC content.
</p>
</div>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">tRNAscan_intron</span>
</pre></div>
<div class="figure">
<img src="tRNAscanImport_files/figure-html/plot4-1.png" alt="tRNAs with introns." width="700"><p class="caption">
tRNAs with introns.
</p>
</div>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="kw">plots</span><span class="op">$</span><span class="kw">variableLoop_length</span>
</pre></div>
<div class="figure">
<img src="tRNAscanImport_files/figure-html/plot5-1.png" alt="Length of the variable loop." width="700"><p class="caption">
Length of the variable loop.
</p>
</div>
</div>
<div id="getting-trna-precursor-sequences" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-trna-precursor-sequences" class="anchor"></a>Getting tRNA precursor sequences</h1>
<p>Since tRNAscan reports the genomic location for tRNAs found, approximate tRNA precursor sequences can be retrieved by combining a tRNAscan input object with matching genomic sequences for the function <code>get.tRNAprecursor</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">BSgenome.Scerevisiae.UCSC.sacCer3</span>)
<span class="kw">genome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html">getSeq</a></span>(<span class="kw">BSgenome.Scerevisiae.UCSC.sacCer3</span>)
<span class="co"># renaming chromosome to match tRNAscan output</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">genome</span>) <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="kw">genome</span>)[<span class="op">-</span><span class="fl">17L</span>],<span class="st">"chrmt"</span>)
<span class="kw">tRNAprecursor</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get.tRNAprecursor.html">get.tRNAprecursor</a></span>(<span class="kw">gr</span>, <span class="kw">genome</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">tRNAprecursor</span>)
</pre></div>
<pre><code>## DNAStringSet object of length 6:
##     width seq                                               names               
## [1]   203 CAATTTGTATATATATACATCTA...AATTAAAGTAGCAGTACTTCAAC pre_chrI.tRNA1
## [2]   173 AGCTTCTAAGCACTTACCATTCC...AATTCGTGAATAGCTGACTGTCA pre_chrI.tRNA2
## [3]   214 GTCAGTGTCCAAATAGTTAAAAC...CATAATCTACGTAGGAATGAAAG pre_chrI.tRNA3
## [4]   182 GTCATACTGACATATCTCATTTT...CCTCTTCAAAGCATACTCATCTT pre_chrI.tRNA4
## [5]   184 GGGTAAAATAGGGTATTTAACTG...ATTAACTAGAATAATAGGGAAAT pre_chrII.tRNA1
## [6]   191 TTTGCTAATAATAAATCTATTTC...TTCATTTCTAGGCCTGTTTCTCC pre_chrII.tRNA2</code></pre>
<p>The length of the overhangs can be defined with the arguments <code>add.5prime</code> and <code>add.3prime</code>, respectively. Both support individual lengths for each tRNA and require values to be integer only. In addition, introns can be removed by setting <code>trim.introns = TRUE</code>.</p>
</div>
<div id="further-reading" class="section level1">
<h1 class="hasAnchor">
<a href="#further-reading" class="anchor"></a>Further reading</h1>
<p>Further examples of working with tRNA information can be found in the <a href="tRNA.html">vignette</a> of the <code>tRNA</code> package.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
</pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so
## 
## locale:
##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] BSgenome.Scerevisiae.UCSC.sacCer3_1.4.0
##  [2] BSgenome_1.56.0                        
##  [3] rtracklayer_1.48.0                     
##  [4] tRNAscanImport_1.9.1                   
##  [5] tRNA_1.6.0                             
##  [6] Structstrings_1.4.0                    
##  [7] Biostrings_2.56.0                      
##  [8] XVector_0.28.0                         
##  [9] GenomicRanges_1.40.0                   
## [10] GenomeInfoDb_1.24.2                    
## [11] IRanges_2.22.2                         
## [12] S4Vectors_0.26.1                       
## [13] BiocGenerics_0.34.0                    
## [14] BiocStyle_2.16.0                       
## 
## loaded via a namespace (and not attached):
##  [1] Biobase_2.48.0               assertive.sets_0.0-3        
##  [3] assertthat_0.2.1             assertive.data_0.0-3        
##  [5] highr_0.8                    BiocManager_1.30.10         
##  [7] GenomeInfoDbData_1.2.3       Rsamtools_2.4.0             
##  [9] yaml_2.2.1                   pillar_1.4.6                
## [11] backports_1.1.8              lattice_0.20-41             
## [13] glue_1.4.1                   assertive.data.uk_0.0-2     
## [15] assertive.matrices_0.0-2     digest_0.6.25               
## [17] assertive.types_0.0-3        RColorBrewer_1.1-2          
## [19] colorspace_1.4-1             htmltools_0.5.0             
## [21] Matrix_1.2-18                XML_3.99-0.4                
## [23] pkgconfig_2.0.3              assertive.data.us_0.0-2     
## [25] assertive.properties_0.0-4   assertive.reflection_0.0-4.1
## [27] bookdown_0.20                zlibbioc_1.34.0             
## [29] scales_1.1.1                 assertive_0.3-5.1           
## [31] BiocParallel_1.22.0          tibble_3.0.3                
## [33] farver_2.0.3                 ggplot2_3.3.2               
## [35] ellipsis_0.3.1               withr_2.2.0                 
## [37] SummarizedExperiment_1.18.2  assertive.code_0.0-3        
## [39] magrittr_1.5                 crayon_1.3.4                
## [41] assertive.strings_0.0-3      memoise_1.1.0               
## [43] evaluate_0.14                Modstrings_1.4.0            
## [45] fs_1.4.2                     assertive.numbers_0.0-2     
## [47] tools_4.0.2                  lifecycle_0.2.0             
## [49] matrixStats_0.56.0           assertive.files_0.0-2       
## [51] stringr_1.4.0                munsell_0.5.0               
## [53] DelayedArray_0.14.1          compiler_4.0.2              
## [55] pkgdown_1.5.1.9000           rlang_0.4.7                 
## [57] grid_4.0.2                   RCurl_1.98-1.2              
## [59] assertive.models_0.0-2       assertive.base_0.0-7        
## [61] bitops_1.0-6                 labeling_0.3                
## [63] rmarkdown_2.3.2              gtable_0.3.0                
## [65] codetools_0.2-16             assertive.datetimes_0.0-2   
## [67] R6_2.4.1                     GenomicAlignments_1.24.0    
## [69] knitr_1.29                   rprojroot_1.3-2             
## [71] desc_1.2.0                   stringi_1.4.6               
## [73] vctrs_0.3.2                  xfun_0.15</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Chan-2016">
<p>Chan, Patricia P., and Todd M. Lowe. 2016. “GtRNAdb 2.0: An Expanded Database of Transfer Rna Genes Identified in Complete and Draft Genomes.” <em>Nucleic Acids Research</em> 44 (D1): D184–9. <a href="https://doi.org/10.1093/nar/gkv1309">https://doi.org/10.1093/nar/gkv1309</a>.</p>
</div>
<div id="ref-Lowe-1997">
<p>Lowe, T. M., and S. R. Eddy. 1997. “TRNAscan-Se: A Program for Improved Detection of Transfer Rna Genes in Genomic Sequence.” <em>Nucleic Acids Research</em> 25 (5): 955–64.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Felix G.M. Ernst.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
